wrangler/src/trust/trust.js

```diff
@@ -0,0 +1,271 @@
+    return accountinfo.sequence
+}
+/**
+ * This function checks the local toml to see if the turret address is trusted
+ * @param turret - the address of the turret you want to check for trust
+ * @returns A boolean value.
+ */
+export async function checkLocalQuorem(turret) {
+    const localquorem = localtoml()
+    try{
+        const thetoml = parse(localquorem).TSS.TURRETS
+        const quorem = []
+        for (const each in thetoml){
+            quorem.push(thetoml[each])
+        }
+        if(quorem.includes(turret))
+            return true
+        return false
+    }catch(err){
+        throw(`the local toml had a problem ${err}`)
+    }
+}
+
+/**
+ * The heal function is used to add or remove a signer from a source account
+ * @param sourceaccount - the account that the signer is being swapped on
+ * @param oldturret - the address of the turret that is being removed from the source account
+ * @param newturret - the new turret to add to the source account
+ * @param functionhash - the hash of the function that is being healed
+ * @returns The transaction XDR, the signer public key, and the signature.
+ */
+export async function heal(sourceaccount, oldturret, newturret, functionhash) {
+    try{
+        /* check that the turret is not trying to heal itself */
+        if (newturret === TURRET_ADDRESS || oldturret === TURRET_ADDRESS)
+            throw ('A Turret may not add or remove itself to a control account')
+        /* gather source account signers */
+        const sourceaccountsigners = await getSignersOnAccount(sourceaccount)
+        /* find the local signer for given functionhash */
+        const localsigner = await getLocalFunctionSigner(functionhash)
+        /* verify the local signer is a signer on the source account */
+        if(!sourceaccountsigners.includes(localsigner))
+            throw (`This turret is not a signer on ${functionhash}, what are you doing?`)
+        /* gather the signer/turret managedata entries from the source account */
+        const sourceaccountmanagedatasigners = await getSourceAccountSignerDataEntries(sourceaccount)
+        /* check that the manage data does not yet include the new turret */
+        if (sourceaccountmanagedatasigners.turret.includes(newturret))
+            throw (`The New turret is already listed in the managedata entries for account`)
+        /* check that the oldturret is listed on the source account managedata entries */
+        if (!sourceaccountmanagedatasigners.turret.includes(oldturret))
+            throw (`The old turret is not listed as a signer in the manage data entries`)
+        /* get the signer address to remove from the managedata entries on the source account */
+        const signertoremoveindex = sourceaccountmanagedatasigners.turret.findIndex(checkaddr);
+        const removesigner = sourceaccountmanagedatasigners.signer[signertoremoveindex]
+        function checkaddr(key) {
+            return key === oldturret;
+        }
```

Comment:
const signertoremoveindex = sourceaccountmanagedatasigners.turret.findIndex((signer) => signer === oldturret);

would allow us to get rid of `checkaddr` function